Gems/Atom/Feature/Common/Code/Tests/IndexedDataVectorTests.cpp

```diff
@@ -8,338 +8,218 @@
 
 #include <AzCore/UnitTest/TestTypes.h>
 #include <AzCore/Component/ComponentApplication.h>
-#include <Atom/Feature/Utils/MultiIndexedDataVector.h>
+#include <Atom/Feature/Utils/IndexedDataVector.h>
 #include <AzCore/Memory/SystemAllocator.h>
 #include <gtest/gtest.h>
 
-
 namespace UnitTest
 {
     using namespace AZ;
     using namespace AZ::Render;
-
+    
     class IndexedDataVectorTests
-        : public ::testing::Test
+        : public UnitTest::AllocatorsTestFixture
     {
     public:
         void SetUp() override
         {
-            CreateAllocator();
+            UnitTest::AllocatorsTestFixture::SetUp();
         }
 
         void TearDown() override
         {
-            DestroyAllocator();
+            UnitTest::AllocatorsTestFixture::TearDown();
         }
-        
-    private:
 
-        void CreateAllocator()
+        template<typename T>
+        IndexedDataVector<T> SetupIndexedDataVector(size_t size, T initialValue = T(0), T incrementAmount = T(1), AZStd::vector<uint16_t>* indices = nullptr)
         {
-            static constexpr size_t NumMBToAllocate = 1;
-            SystemAllocator::Descriptor desc;
-            desc.m_heap.m_numFixedMemoryBlocks = 1;
-            desc.m_heap.m_fixedMemoryBlocksByteSize[0] = NumMBToAllocate * 1024 * 1024;
-            m_memBlock = AZ_OS_MALLOC(
-                desc.m_heap.m_fixedMemoryBlocksByteSize[0],
-                desc.m_heap.m_memoryBlockAlignment);
-            desc.m_heap.m_fixedMemoryBlocks[0] = m_memBlock;
-
-            AllocatorInstance<AZ::SystemAllocator>::Create(desc);
+            IndexedDataVector<T> data;
+            T value = initialValue;
+            for (size_t i = 0; i < size; ++i)
+            {
+                uint16_t index = data.GetFreeSlotIndex();
+                EXPECT_NE(index, IndexedDataVector<int>::NoFreeSlot);
+                if (indices)
+                {
+                    indices->push_back(index);
+                }
+                if (index != IndexedDataVector<int>::NoFreeSlot)
+                {
+                    data.GetData(index) = value;
+                    value += incrementAmount;
+                }
+            }
+            return data;
         }
 
-        void DestroyAllocator()
+        template<typename T>
+        void ShuffleIndexedDataVector(IndexedDataVector<T> dataVector, AZStd::vector<uint16_t> indices)
```

Comment:
I think you meant to pass dataVector by reference. It's currently passed by value so this function doesn't actually change anything.

Reply:
Good catch. Thankfully the unit tests still worked :)